### PR TITLE
Adding SSL training functionality

### PR DESCRIPTION
**Pull request by kvantricht**
PR includes:
- First SSL training loop 
- Update to Presto wrapper to include a `SSLPresto` model as well, with a SSL forward pass returning the loss, reconstructed image, and mask
- Simple test for SSL training loop on Presto added

Still needs validation on whether masking happens correctly and training loop is implemented correctly.

### CODE REVIEW

**Comment by kvantricht** [comment]
Naively testing with 10-day dataset, by doing:

`mask_params = MaskParamsNoDw(
            strategies=MASK_STRATEGIES, ratio=0.75, num_timesteps=36
        )`

gives a shape mismatch, so that requires further investigation.

**Comment by kvantricht** [comment]
No longer relevant for the time being.
